generator/internal/parser/httprule/http_rule_parser.go

```diff
@@ -0,0 +1,191 @@
+		}
+
+	}
+}
+
+// ### Path template syntax
+//
+//	Template = "/" Segments [ Verb ] ;
+//	Segments = Segment { "/" Segment } ;
+//	Segment  = "*" | "**" | LITERAL | Variable ;
+//	Variable = "{" FieldPath [ "=" Segments ] "}" ;
+//	FieldPath = IDENT { "." IDENT } ;
+//	Verb     = ":" LITERAL ;
+const (
+	segmentError segmentType = iota // 0
+	segmentLiteral
+	segmentIdentifier
+	segmentVerb //a verb is actually ':' + LITERAL,
+	// but we need a way to differentiate a literal within a segment, from a literal within a verb
+	segmentEOF
+)
+
+const (
+	slash    = '/'
+	star     = '*'
+	varLeft  = '{'
+	varRight = '}'
+	varSep   = '='
+	verbSep  = ':'
+)
+
+type stateFn func(*lexer) stateFn
+
+// slashState is the first state before a segment
+func slashState(l *lexer) stateFn {
+	if !l.ignoreIfMatches(slash) {
+		return l.unexpectedRuneError(slash, l.peek())
+	}
+	return segmentState
+}
+
+func segmentState(l *lexer) stateFn {
+	switch l.peek() {
+	case eof, slash, verbSep:
+		return l.errorf("expected a segment, found %q", l.peek())
+	case varLeft:
+		return varState
+	}
+	return literalState
+}
+
+const validFirstIdentifierRunes = "ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+	"abcdefghiljkmnopqrstuvwxyz"
+
+const validIdentifierRunes = validFirstIdentifierRunes +
+	"0123456789" +
+	"_"
+const validVariableValueRunes = validIdentifierRunes + "/*"
+
+func varState(l *lexer) stateFn {
+	if !l.ignoreIfMatches(varLeft) {
+		return l.unexpectedRuneError(varLeft, l.peek())
+	}
+
+	if !l.accept(validFirstIdentifierRunes) {
+		return l.errorf("expected a valid first rune for a variable identifier, got %q", l.peek())
+	}
+
+	// The identifier is a single or multiple characters, just take all that match.
+	_ = l.acceptAll(validIdentifierRunes)
+
+	// We have reached the end of the valid characters for an identifier
+	// emit the segment now and process the rest of the variable for correctness.
+	l.emit(segmentIdentifier)
+
+	if l.ignoreIfMatches(varSep) {
+		// This var has a value, we don't use it for anything, so lets just ignore it.
+		//TODO The valid format for a variable value is a sequence of segments, we could validate that here.
+		if !l.acceptAll(validVariableValueRunes) {
+			return l.errorf("expected a valid variable value, got %q", l.peek())
+		}
+		l.ignore()
+	}
+
+	if !l.ignoreIfMatches(varRight) {
+		return l.unexpectedRuneError(varRight, l.peek())
+	}
+	return eoSegmentState
+}
+
+// verbState must start with a verbSep
+func verbState(l *lexer) stateFn {
+	if !l.ignoreIfMatches(verbSep) {
+		return l.unexpectedRuneError(verbSep, l.peek())
+	}
+	if !l.acceptAll(validLiteralRunes) {
+		return l.errorf("expected a literal segment")
+	}
+	l.emit(segmentVerb)
+	// the only valid rune after a verb is EOF
+	return eofState
+}
+
+const validLiteralRunes = "abcdefghijklmnopqrstuvwxyz" +
+	"ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+	"0123456789" +
+	"-._~" +
+	"%" +
+	"!$&'()*+,;="
+
+func literalState(l *lexer) stateFn {
+	if l.accept(string(star)) {
```

Comment:
I am assuming path segments like `/*/` and `/**/` should be parsed into literals, but couldn't extract a definitive answer from the spec.